[DA] Date 타입 적용, attributedString 적용

kimkyunghun3 opened the pull request:
# 개요
- 🔗  이슈링크 : #80 


# 변경사항

## 작업 내용
- Date관련 내용 String -> Date타입으로 변경
- 각 Date타입별 DateFormatter 로직 구현(현재 서버와 연동이 없어서 서버와 연결된 로직은 추후 추가하겠습니다)
- AttributedString으로 일부 Text 색상 적용

### 스크린샷

| 작업전 | 작업후 |
|:---:|:---:|
|<img src ="https://user-images.githubusercontent.com/52434820/182096826-466c9494-297d-4b68-8cef-f6649cb1b7cc.png" width ="200">|<img src ="https://user-images.githubusercontent.com/52434820/182592457-698bfd0d-b88d-4851-9bf5-b6aaeda33291.png" width ="200">|

kimkyunghun3 committed:
:recycle: DateFormatter format 수정 및 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: 유효기간, 마감시간 Date 타입으로 변경에 따른 로직 수정

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:bulb: listCardDeadlineView 일부 String 색상 적용 기능 구현

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

szzang-kr commented on Projects/DesignSystem/Sources/Component/Extension/Date+sugar.swift:
```diff
@@ -9,7 +9,14 @@
 
 extension Date {
-    func convertToString() -> String {
+    func ListCardDeadlineString() -> String {
```
```suggestion
    func hourString() -> String {
```

조금더 일반적인 네이밍이 좋을 것 같아요!

kimkyunghun3 replied:
제안해주신 방향이 좀 더 명확한거같네요! 제안 감사합니다!

szzang-kr commented on Projects/DesignSystem/Sources/Component/Extension/Date+sugar.swift:
```diff
@@ -9,7 +9,14 @@
+        return dateString
+    }
+
+    func listCardString() -> String {
```
마찬가지로 일반적인 네이밍이 필요할것같아요!

kimkyunghun3 replied:
년,월,일을 3개로 변환해주기 때문에 `fullDateString()` 으로 해서 좀 더 명확하게 변경했습니다!

szzang-kr commented on Projects/DesignSystem/Sources/Component/ListCard/DDipListCardDeadlineView.swift:
```diff
@@ -22,8 +22,32 @@ public class DDipListCardDeadlineView: DDipListCardApplyBaseView {
+            range: NSString(string: fullText).range(of: leftTimeText)
+        )
+
+        attributedString.setAttributes([
+            .font: UIFont.designSystem(.pretendard, family: .bold, size: ._14)
+        ], range: NSString(string: fullText).range(of: resultTitle))
```
컨벤션 위쪽과 통일 하면 좋을것 같습니다 ! 😄

kimkyunghun3 replied:
컨벤션 통일되도록 수정했습니다!!!

szzang-kr reviewed (changes requested):
👍

kimkyunghun3 committed:
:recycle: Date -> String 변환 네이밍 명확하게 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: attributedString관련 로직 convention 통일 

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

AhnSangHee reviewed (approved):
고생하셨습니다!👍👍
